cli/command/service_test.go

```diff
@@ -0,0 +1,418 @@
+
+		base.Resolver.EXPECT().
+			Resolve("environment", "env_name").
+			Return([]string{"env_id"}, nil)
+
+		base.Resolver.EXPECT().
+			Resolve("load_balancer", "lb_name").
+			Return([]string{"lb_id"}, nil)
+
+		req := models.CreateServiceRequest{
+			DeployID:       "dpl_id",
+			EnvironmentID:  "env_id",
+			LoadBalancerID: "lb_id",
+			ServiceName:    "svc_name",
+		}
+
+		base.Client.EXPECT().
+			CreateService(req).
+			Return("job_id", nil)
+
+		if wait {
+			job := &models.Job{
+				Status: "Completed",
+				Result: "svc_id",
+			}
+
+			base.Client.EXPECT().
+				ReadJob("job_id").
+				Return(job, nil)
+
+			deployments := []models.Deployment{
+				{
+					DesiredCount: 1,
+					RunningCount: 1,
+				},
+			}
+
+			service := &models.Service{
+				Deployments:  deployments,
+				DesiredCount: 1,
+				RunningCount: 1,
+			}
+
+			base.Client.EXPECT().
+				ReadService("svc_id").
+				Return(service, nil).
+				AnyTimes()
+		}
+
+		args := Args{"env_name", "svc_name", "dpl_name"}
+		flags := Flags{"loadbalancer": "lb_name"}
+		c := NewContext(t, args, flags, SetNoWait(!wait))
+
+		serviceCommand := NewServiceCommand(base.Command())
+		if err := serviceCommand.create(c); err != nil {
+			t.Fatal(err)
+		}
+	})
+}
+
+func TestCreateService_userInputErrors(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	contexts := map[string]*cli.Context{
+		"Missing ENVIRONMENT arg": NewContext(t, nil, nil),
+		"Missing NAME arg":        NewContext(t, Args{"env_name"}, nil),
+		"Missing DEPLOY arg":      NewContext(t, Args{"env_name", "svc_name"}, nil),
+	}
+
+	serviceCommand := NewServiceCommand(base.Command())
+	for name, c := range contexts {
+		t.Run(name, func(t *testing.T) {
+			if err := serviceCommand.create(c); err == nil {
+				t.Fatal("%s: error was nil!", name)
+			}
+		})
+	}
+}
+
+func TestDeleteService(t *testing.T) {
+	testWaitHelper(t, func(t *testing.T, wait bool) {
+		base, ctrl := newTestCommand(t)
+		defer ctrl.Finish()
+
+		base.Resolver.EXPECT().
+			Resolve("service", "svc_name").
+			Return([]string{"svc_id"}, nil)
+
+		base.Client.EXPECT().
+			DeleteService("svc_id").
+			Return("job_id", nil)
+
+		if wait {
+			job := &models.Job{
+				Status: "Completed",
+				Result: "svc_id",
+			}
+
+			base.Client.EXPECT().
+				ReadJob("job_id").
+				Return(job, nil)
+		}
+
+		args := Args{"svc_name"}
+		c := NewContext(t, args, nil, SetNoWait(!wait))
+
+		serviceCommand := NewServiceCommand(base.Command())
+		if err := serviceCommand.delete(c); err != nil {
+			t.Fatal(err)
+		}
+	})
+}
+
+func TestDeleteService_userInputError(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": NewContext(t, nil, nil),
+	}
+
+	serviceCommand := NewServiceCommand(base.Command())
+	for name, c := range contexts {
+		t.Run(name, func(t *testing.T) {
+			if err := serviceCommand.create(c); err == nil {
+				t.Fatal("%s: error was nil!", name)
```

Comment:
[nit] we don't need to show the case name anymore in the error since `t.Run(name` does that for us

Reply:
Mm, yep yep. ✅